components/ieee802154/driver/esp_ieee802154_dev.c

```diff
@@ -343,10 +343,8 @@ static void enable_rx(void)
 {
     set_next_rx_buffer();
     IEEE802154_SET_TXRX_PTI(IEEE802154_SCENE_RX);
-
-    ieee802154_set_cmd(IEEE802154_CMD_RX_START);
-
     ieee802154_set_state(IEEE802154_STATE_RX);
+    ieee802154_set_cmd(IEEE802154_CMD_RX_START);
```

Comment:
Here the enable_rx will be always running in critical context or ISR context. So there will be no 154 interrupt which can break the processing of this function. Also no need to consider the order of these two lines.